File: epi_judge_python/delete_from_list.py
import functools

from test_framework import generic_test
from test_framework.test_utils import enable_executor_hook


# Delete the node past this one. Assume node is not a tail.
def delete_after(node):
    # TODO - you fill in here.
    return


@enable_executor_hook
def delete_from_list_wrapper(executor, head, node_idx):
    node_to_delete = head
    prev = None
    if node_to_delete is None:
        raise RuntimeError('List is empty')
    for _ in range(node_idx):
        if node_to_delete.next is None:
            raise RuntimeError("Can't delete last node")
        prev = node_to_delete
        node_to_delete = node_to_delete.next

    executor.run(functools.partial(delete_after, prev))

    return head


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main("delete_from_list.py",
                                       'delete_from_list.tsv',
                                       delete_from_list_wrapper))

File: epi_judge_cpp_solutions/delete_from_list.cc
#include <memory>

#include "list_node.h"
#include "test_framework/generic_test.h"
#include "test_framework/timed_executor.h"

using std::shared_ptr;

// Delete the node past this one. Assume node is not a tail.
void DeleteAfter(const shared_ptr<ListNode<int>>& node) {
  node->next = node->next->next;
}

shared_ptr<ListNode<int>> DeleteFromListWrapper(
    TimedExecutor& executor, const shared_ptr<ListNode<int>>& head,
    int node_idx) {
  shared_ptr<ListNode<int>> selected_node = head;
  shared_ptr<ListNode<int>> prev;
  while (node_idx-- > 0) {
    if (!selected_node || !selected_node->next)
      throw std::runtime_error("Node index is out of range");
    prev = selected_node;
    selected_node = selected_node->next;
  }
  executor.Run([&] { DeleteAfter(prev); });
  return head;
}

int main(int argc, char* argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"executor", "head", "node_idx"};
  return GenericTestMain(args, "delete_from_list.cc", "delete_from_list.tsv",
                         &DeleteFromListWrapper, DefaultComparator{},
                         param_names);
}
